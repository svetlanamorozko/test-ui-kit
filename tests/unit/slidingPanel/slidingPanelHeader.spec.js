import { mount } from 'enzyme';
import React from 'react';
import SlidingPanelHeader from '../../../components/slidingPanel/slidingPanelHeader/slidingPanelHeader';

jest.useFakeTimers();

describe('SlidingPanel', () => {
  describe('#render()', () => {
    it('render noscript without children', () => {
      const renderTree = mount(
        <SlidingPanelHeader />
      );

      jest.runAllTimers();

      expect(renderTree).toMatchSnapshot();
    });

    it('render with children', () => {
      const title = 'Test Title';
      const renderTree = mount(
        <SlidingPanelHeader>
          { title }
        </SlidingPanelHeader>
      );

      const titleEl = renderTree.find('.ui-sliding-panel-header__title');
      const defaultRightBlock = renderTree.find('.ui-sliding-panel-header__close-button');

      jest.runAllTimers();

      expect(renderTree).toMatchSnapshot();
      expect(titleEl.text()).toEqual(title);
      expect(defaultRightBlock.text()).toEqual('×');
    });

    it('render with custom left and right blocks', () => {
      const title = 'Test Title';
      const renderTree = mount(
        <SlidingPanelHeader
          renderLeftBlock={
            props => (
              <button
                {...props}
                style={{ marginLeft: '15px' }}
              > ← </button>
            )
          }
          renderRightBlock={
            props => (
              <button
                {...props}
                data-rel="close"
                style={{ marginRight: '15px' }}
              > close me! </button>
            )
          }
        >
          {title}
        </SlidingPanelHeader>
      );

      const titleEl = renderTree.find('.ui-sliding-panel-header__title');

      jest.runAllTimers();

      expect(renderTree).toMatchSnapshot();
      expect(titleEl.text()).toEqual(title);
    });

    it('render with title as a node', () => {
      const backButtonText = ' ← ';
      const closeButtonText = ' close me! ';
      const title = <span className="my-class"> my label </span>;
      const renderTree = mount(
        <SlidingPanelHeader
          renderLeftBlock={
            props => (
              <button
                {...props}
                style={{ marginLeft: '15px' }}
              >{backButtonText}</button>
            )
          }
          renderRightBlock={
            props => (
              <button
                {...props}
                data-rel="close"
                style={{ marginRight: '15px' }}
              >{closeButtonText}</button>
            )
          }
        >
          { title }
        </SlidingPanelHeader>
      );

      const titleEl = renderTree.find('.ui-sliding-panel-header__title');
      const titleContent = renderTree.find('.my-class');
      const rightBlock = renderTree.find('.ui-sliding-panel-header__close-button');
      const leftBlock = renderTree.find('.ui-sliding-panel-header__left-block-back');

      jest.runAllTimers();

      expect(titleEl).toHaveLength(1);
      expect(titleContent).toHaveLength(1);
      expect(rightBlock.text()).toEqual(closeButtonText);
      expect(leftBlock.text()).toEqual(backButtonText);
    });

    it('should render with backButtonLabel and onBackButtonClick', () => {
      const title = <span className="my-class"> my label </span>;
      const backButtonClick = jest.fn();
      const renderTree = mount(
        <SlidingPanelHeader
          backButtonLabel="foo"
          onBackButtonClick={backButtonClick}
          useDefaultLeftBlock
        >
          { title }
        </SlidingPanelHeader>
      );

      expect(renderTree).toMatchSnapshot();
    });

    it('should render with dataAttr and className', () => {
      const title = <span className="my-class"> my label </span>;
      const renderTree = mount(
        <SlidingPanelHeader
          className="foo"
          dataAttrs={{ foo: 'bar' }}
        >
          { title }
        </SlidingPanelHeader>
      );

      expect(renderTree).toMatchSnapshot();
    });

    it('should render with deprecated rightBlock and leftBlock', () => {
      const backButtonText = ' ← ';
      const closeButtonText = ' close me! ';
      const title = <span className="my-class"> my label </span>;
      const renderTree = mount(
        <SlidingPanelHeader
          leftBlock={<button
            className="back-button"
            data-rel="close"
            style={{ marginRight: '15px' }}
          >{backButtonText}</button>}
          rightBlock={<button
            className="close-button"
            data-rel="close"
            style={{ marginRight: '15px' }}
          >{closeButtonText}</button>}
        >
          { title }
        </SlidingPanelHeader>
      );

      const rightBlock = renderTree.find('.close-button');
      const leftBlock = renderTree.find('.back-button');

      jest.runAllTimers();

      expect(renderTree).toMatchSnapshot();

      expect(rightBlock.text()).toEqual(closeButtonText);
      expect(leftBlock.text()).toEqual(backButtonText);
    });
  });
});
