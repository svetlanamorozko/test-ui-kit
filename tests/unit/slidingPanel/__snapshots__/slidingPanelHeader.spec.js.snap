// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`SlidingPanel #render() render noscript without children 1`] = `<SlidingPanelHeader />`;

exports[`SlidingPanel #render() render with children 1`] = `
<SlidingPanelHeader>
  <div
    className="ui-sliding-panel-header"
  >
    <div
      className="ui-sliding-panel-header__left-block"
    />
    <h3
      className="ui-sliding-panel-header__title"
    >
      Test Title
    </h3>
    <div
      className="ui-sliding-panel-header__right-block"
    >
      <button
        className="ui-sliding-panel-header__close-button"
        data-rel="close"
      >
        ×
      </button>
    </div>
  </div>
</SlidingPanelHeader>
`;

exports[`SlidingPanel #render() render with custom left and right blocks 1`] = `
<SlidingPanelHeader
  renderLeftBlock={[Function]}
  renderRightBlock={[Function]}
>
  <div
    className="ui-sliding-panel-header"
  >
    <div
      className="ui-sliding-panel-header__left-block"
    >
      <button
        className="ui-sliding-panel-header__left-block-back"
        style={
          Object {
            "marginLeft": "15px",
          }
        }
      >
         ← 
      </button>
    </div>
    <h3
      className="ui-sliding-panel-header__title"
    >
      Test Title
    </h3>
    <div
      className="ui-sliding-panel-header__right-block"
    >
      <button
        className="ui-sliding-panel-header__close-button"
        data-rel="close"
        style={
          Object {
            "marginRight": "15px",
          }
        }
      >
         close me! 
      </button>
    </div>
  </div>
</SlidingPanelHeader>
`;

exports[`SlidingPanel #render() should render with backButtonLabel and onBackButtonClick 1`] = `
<SlidingPanelHeader
  backButtonLabel="foo"
  onBackButtonClick={[MockFunction]}
  useDefaultLeftBlock={true}
>
  <div
    className="ui-sliding-panel-header"
  >
    <div
      className="ui-sliding-panel-header__left-block"
    >
      <button
        className="ui-sliding-panel-header__left-block-back"
        onClick={[MockFunction]}
      >
        <span
          className="ui-sliding-panel-header__left-block-back-icon"
        />
        <span
          className="ui-sliding-panel-header__left-block-back-text"
        >
          foo
        </span>
      </button>
    </div>
    <h3
      className="ui-sliding-panel-header__title"
    >
      <span
        className="my-class"
      >
         my label 
      </span>
    </h3>
    <div
      className="ui-sliding-panel-header__right-block"
    >
      <button
        className="ui-sliding-panel-header__close-button"
        data-rel="close"
      >
        ×
      </button>
    </div>
  </div>
</SlidingPanelHeader>
`;

exports[`SlidingPanel #render() should render with dataAttr and className 1`] = `
<SlidingPanelHeader
  className="foo"
  dataAttrs={
    Object {
      "foo": "bar",
    }
  }
>
  <div
    className="ui-sliding-panel-header foo"
    data-foo="bar"
  >
    <div
      className="ui-sliding-panel-header__left-block"
    />
    <h3
      className="ui-sliding-panel-header__title"
    >
      <span
        className="my-class"
      >
         my label 
      </span>
    </h3>
    <div
      className="ui-sliding-panel-header__right-block"
    >
      <button
        className="ui-sliding-panel-header__close-button"
        data-rel="close"
      >
        ×
      </button>
    </div>
  </div>
</SlidingPanelHeader>
`;

exports[`SlidingPanel #render() should render with deprecated rightBlock and leftBlock 1`] = `
<SlidingPanelHeader
  leftBlock={
    <button
      className="back-button"
      data-rel="close"
      style={
        Object {
          "marginRight": "15px",
        }
      }
    >
       ← 
    </button>
  }
  rightBlock={
    <button
      className="close-button"
      data-rel="close"
      style={
        Object {
          "marginRight": "15px",
        }
      }
    >
       close me! 
    </button>
  }
>
  <div
    className="ui-sliding-panel-header"
  >
    <div
      className="ui-sliding-panel-header__left-block"
    >
      <button
        className="back-button"
        data-rel="close"
        style={
          Object {
            "marginRight": "15px",
          }
        }
      >
         ← 
      </button>
    </div>
    <h3
      className="ui-sliding-panel-header__title"
    >
      <span
        className="my-class"
      >
         my label 
      </span>
    </h3>
    <div
      className="ui-sliding-panel-header__right-block"
    >
      <button
        className="close-button"
        data-rel="close"
        style={
          Object {
            "marginRight": "15px",
          }
        }
      >
         close me! 
      </button>
    </div>
  </div>
</SlidingPanelHeader>
`;
